src/LaravelCommon.php

```diff
@@ -2,12 +2,32 @@
 
 namespace LiveIntent\LaravelCommon;
 
+use Illuminate\Auth\RequestGuard;
+use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\Queue;
 use Illuminate\Support\Facades\Route;
 use Illuminate\Foundation\Http\Events\RequestHandled;
+use LiveIntent\LaravelCommon\Auth\Guards\LITokenGuard;
 
 class LaravelCommon
 {
+    /**
+     * Register Tessellate token authentication guard
+     */
+    public static function authGuard()
+    {
+        Auth::extend('li_token', function ($app, $_name, array $config) {
+            throw_if($config['provider'] == null);
```

Comment:
looks fancy but because you didn't specify the second argument, I wonder if it will provide us enough info when it needs to throw an exception?

Reply:
I'd actually probably leave this throw out honestly, I think it's not really needed and may actually hide the real problem that we'd otherwise be able to determine from the error message that was thrown. (Also this very well could be part of the infinite loop problem :) )

Reply:
Sure, will remove this piece. I think you may be right that this may be the culprit for the infinite loop, but will test that piece.